### PR TITLE
druntime: adjust LoongArch64 definitions

### PR DESCRIPTION
**Pull request by scylaac**

### CODE REVIEW

File: druntime/src/core/thread/fiber.d [hunk 1]
```diff
@@ -1467,6 +1450,32 @@ private:
              */
             pstack += int.sizeof * 1;
         }
+        else version (AsmLoongArch64_Posix)
```
Comment (thewilsonator) [inline druntime/src/core/thread/fiber.d:1453]: please keep this in the same spot (after `AsmMIPS_O32_Posix`) so that its is easier to review

File: druntime/src/core/sys/posix/sys/socket.d [hunk 2]
```diff
@@ -556,6 +556,7 @@ version (linux)
             SO_RCVLOWAT     = 18,
             SO_RCVTIMEO     = 20,
             SO_REUSEADDR    = 2,
+            SO_REUSEPORT    = 15,
```
Comment (thewilsonator) [inline druntime/src/core/sys/posix/sys/socket.d:559]: this appears to be unrelated to `LoongArch64`

File: druntime/src/core/sys/posix/fcntl.d [hunk 3]
```diff
@@ -414,11 +414,10 @@ version (linux)
...
         enum O_DIRECT       = 0x004000; // octal    040000
-        version (D_LP64)
-            enum O_LARGEFILE = 0;
-        else
-            enum O_LARGEFILE = 0x8000;  // octal   0100000
-        enum O_TMPFILE      = 0x404000; // octal 020040000
+
+        enum O_LARGEFILE = 0;
+
+        enum O_TMPFILE      = 0x410000; // octal 020200000
```
Comment (thewilsonator) [inline druntime/src/core/sys/posix/fcntl.d:420]: ditto

File: druntime/src/core/sys/posix/sys/socket.d (hunk 2 above)
Comment (scylaac) [reply druntime/src/core/sys/posix/sys/socket.d:559]: Thanks for the review!

I am not an expert in this... Could you let me know why is it so (and the one below)?  I simply copied this definitions from the kernel headers with:

```gcc -E -P -include sys/socket.h - <<< "SO_REUSEPORT"```

File: druntime/src/core/sys/posix/fcntl.d (hunk 3 above)
Comment (scylaac) [reply druntime/src/core/sys/posix/fcntl.d:420]: and

```gcc -E -P -D_GNU_SOURCE -include fcntl.h - <<< "O_TMPFILE"```

File: druntime/src/core/sys/posix/sys/socket.d (hunk 2 above)
Comment (ibuclaw) [reply druntime/src/core/sys/posix/sys/socket.d:559]: If it's not in the posix standard, then it won't be included here, rather in the Linux specific system socket module.

File: druntime/src/core/sys/posix/fcntl.d (hunk 3 above)
Comment (ibuclaw) [reply druntime/src/core/sys/posix/fcntl.d:420]: Preprocessing headers doesn't give an accurate representation of what the definition should be. There are systems which have been compiled with LARGEFILE.

File: druntime/src/core/sys/posix/sys/socket.d (hunk 2 above)
Comment (scylaac) [reply druntime/src/core/sys/posix/sys/socket.d:559]: Oh, I see. Thanks.

File: druntime/src/core/sys/posix/fcntl.d (hunk 3 above)
Comment (scylaac) [reply druntime/src/core/sys/posix/fcntl.d:420]: I assume `version (LoongArch64)` implies `version (D_LP64)`. So does the other case seem unnecessary?

**Comment by scylaac** [comment]
Removing changes to `core/sys/posix`.

**Review by thewilsonator — APPROVED** [review]
